rosmon_core/src/monitor/node_monitor.cpp

```diff
@@ -174,11 +175,27 @@ void NodeMonitor::start()
 
 	if(m_launchNode->coredumpsEnabled() && g_coreIsRelative && m_processWorkingDirectory.empty())
 	{
-		char tmpfile[256];
-		strncpy(tmpfile, "/tmp/rosmon-node-XXXXXX", sizeof(tmpfile));
-		tmpfile[sizeof(tmpfile)-1] = 0;
-		m_processWorkingDirectory = mkdtemp(tmpfile);
-		m_processWorkingDirectoryCreated = true;
+		if (const char* logDir = std::getenv("ROSMON_LOG_PATH")) 
```

Comment:
Why env var here?

Reply:
It seems like this part of code is called again during the core dump. I don't know how you'd be able to pass the directory through as a variable ?

Reply:
make a class variable and pass in the constructor or in init function maybe.